.Documentation/articles/developer/07-run-a-course.md

```diff
@@ -1,3 +1,93 @@
+# Create a Course
+
+## Entity Factory
+
+The `EntityFactory` is a helper class that allows you to dynamically create basic elements from a course. It is part of the Innoactive Creator's API.
+
+It supports the creation of steps (`IStep`), transitions (`ITransition`), chapters (`IChapter`), and courses (`ICourse`). A course can be edited as much as possible only if it is not being played.
+
+Additionally, the `EntityFactory` provides a built-in post-processing stack for the recently built elements.
+
+## Usage
+
+It is possible to easely create a course element with a single line of code, adding corresponding element to where it belongs also takes a single line, which makes courses creation very flexible and easy to execute:
+
+```csharp
+using UnityEngine;
+using Innoactive.Creator.Core;
+
+// Unity components' classes should have the same names as their files.
+public class TrainingCourseBuilder : MonoBehaviour
+{
+    public ICourse CreateDefaultCourse()
+    {
+        // Create a new ICourse.
+        ICourse course = EntityFactory.CreateCourse("New Course");
+        
+        // Create a new IChapter.
+        IChapter chapter = EntityFactory.CreateChapter("Chapter 1");
+        
+        // Create a new IStep.
+        IStep step = EntityFactory.CreateStep("New Step");
+        
+        // Create a new ITransition.
+        ITransition transition = EntityFactory.CreateTransition();
+        
+        // Adding an ITransition to an IStep.
+        step.Data.Transitions.Data.Transitions.Add(transition);
+        
+        // Adding an IStep to an IChapter.
+        chapter.Data.Steps.Add(step);
+        
+        // Adding an IChapter to an ICourse.
+        course.Data.Chapters.Add(chapter);
+
+        return course;
+    }
+}
+```
+
+By default, all `Istep` and `IChapter` created with the `EntityFactory` already contain a default `ITransition` and `IChapter` respectively. For reference purposes, in the example above, we intentionally show how to create and add them to a course.
+
+## Entity PostProcessing
+
+In the case where it is required to execute certain actions to every new entity, this can be achieved by extending the `EntityPostProcessing <T>` class, T corresponds to the type of the element we aim to post-process.
+
+The post-processing stack is automatically fired by the `EntityFactory` every time it creates a course element. 
```

Comment:
Every time the `EntityFactory` creates a course element, it executes the post-processing stack of actions.